completion_aggregator/tasks/aggregation_tasks.py

```diff
@@ -201,18 +220,47 @@ def update_for_aggregator(self, block, affected_aggregators, force):
             if modified is not None:
                 last_modified = max(last_modified, modified)
         if self._aggregator_needs_update(block, last_modified, force):
-            obj, _ = Aggregator.objects.submit_completion(
-                user=self.user,
-                course_key=self.course_key,
-                block_key=block,
-                aggregation_name=block.block_type,
-                earned=total_earned,
-                possible=total_possible,
-                last_modified=last_modified,
-            )
-            self.aggregators[obj.block_key] = obj
+
+            self._is_aggregator_data_valid(self.user, self.course_key, block)
+            last_modified_unaware = make_datetime_timezone_unaware(last_modified)
+            created_at_unaware = make_datetime_timezone_unaware(timezone.now())
+            updated_at_unaware = make_datetime_timezone_unaware(timezone.now())
+            percent = get_percent(total_earned, total_possible)
+            self.processed_list.append({
+                'user': self.user,
+                'course_key': self.course_key,
+                'block_key': block,
+                'aggregation_name': block.block_type
+            })
+            # Param order: (user_id, course_key, block_key, block_type, earned, possible, percent,
```

Comment:
I know this pattern is used in the file but not sure I like it. It is bypassing model level validation, the internal model details are being exposed here and it cannot be tested in isolation. How about this:

1. We create a Aggregator instance (without saving it) and add it to the list.
2. The Aggregator model can have a method `values()` which will return a dictionary of values of its fields. Similar to https://docs.djangoproject.com/en/1.11/ref/models/querysets/#values. 
3. It can be called in `save_updated_aggregators()` to collect the data from the model instances.
4. In the query, can we not use keyed arguments rather than positional? https://docs.djangoproject.com/en/1.11/topics/db/sql/#passing-parameters-into-raw

Reply:
I have given this a try. Maybe variable names need to be updated a bit.